### DISCUSSION TITLE
layout per tag

### DISCUSSION THREAD

**Discussion by patrikha** [leftwm/leftwm#764] [Ideas]
Hi,

would it be possible to change so that layouts are based on tag instead of workspace. Having multiple screens where the main big center is the main working screen/workspace and switching between tags would help a lot if the layout is preserved on each tag.

Perhaps a config setting to set tag or workspace layout?

thanks
Patrik

**Comment by kid** [comment]
Hello Adventurer,

In this case I think what you want is layouts per workspace, which is supported, it does however requires you to define the workspace with screen size and position in the config, which then allows you to define a list of layouts per screen.

In my case, I have an ultra wide monitor on the center, and a vertical screen on the side, and thus use completely different layouts for each

One downside is when you use a laptop and you frequently disconnect the external screen, things will not work properly because Leftwm will continue using the workspace sizes/position in the config

**Comment by patrikha** [comment]
well not exactly, I have a similar setup but with one vertical screen on each side.
My workflow is that I usually use tag 4 for my Emacs and console windows in a specific layout on my center screen and then switching to tab 8 for my browser and email in another layout (at least when using i3) but now I cant really have different layouts per tag for my center screen when switching tags. Or perhaps I misunderstood your comment!?

**Reply by hertg** [reply]
Your proposal sounds interesting, maybe you can open it as an issue.

**Comment by kid** [comment]
Ok, in that case, I think you're actually looking for `layout_mode = "Tag"`. Not sure how well that works with different workspaces though

**Comment by NicTanghe** [comment]
Hello, I’d also like layouts per tag to be implemented.
As I’d like to have them change depending on the purpose of the programs it's running and not depending on what screen it is visible on.

If anyone knows a way to set this up using workspaces, I`d love to c an example config.

**Comment by VuiMuich** [comment]
For this feature I see two general directions how to approach this config option:
- set the initial layout for a tag (does not apply when a previous state is restored)
- pin the available layout for a tag
Besides creating two different config options for this, I also can imagine a way that kind of unifies this, but is probably a bit more work to implement:
Add a config option `tag_layout_overrides`, which is an array of tuples `("tag_id", ["layout_1", "layout_2", ...])`, if a `tag_id` is not defined by this it falls back to the behaviour we currently have. This would need to be exclusive to `layout_mode: Tag;` obviously.
I took a cursory tour through the layout related code, and it might actually not that much to be added and changed to actually make this possible.

**Reply by NicTanghe** [reply]
The possibility to reorder / change the layouts per tag seems to be the most functional solution 2 me.

**Reply by patrikha** [reply]
I lay down my vote on this since I moved to wayland and hyprland but thanks for the adventurous ride with leftwm :-)

**Comment by grndplane** [comment]
I think PR 1301 might be what your looking for, at least it is a start. It is working good for myself.
